services/s3/src/main/java/software/amazon/awssdk/services/s3/S3Configuration.java

```diff
@@ -167,6 +171,26 @@ public boolean dualstackEnabled() {
         return dualstackEnabled.value();
     }
 
+    /**
+     * Returns whether trailing checksum validation is enabled. This is enabled by default.
```

Comment:
Can we add a note mentioning this performs MD5 checksum validation and we recommended specifying checksumAlgorithm for putObject and enable checksumMode for getObject?